src/blueapi/config.py

```diff
@@ -117,54 +181,48 @@ def __init__(self, schema: type[C]) -> None:
 
     def use_values(self, values: Mapping[str, Any]) -> None:
         """
-        Use all values provided in the config, override any defaults
-        and values set by previous calls into this class.
-
-        Args:
-            values (Mapping[str, Any]): Dictionary of override values,
-                                        does not need to be exhaustive
-                                        if defaults provided.
+        Use all values provided in the config, override any defaults.
         """
-
-        def recursively_update_map(old: dict[str, Any], new: Mapping[str, Any]) -> None:
-            for key in new:
-                if (
-                    key in old
-                    and isinstance(old[key], dict)
-                    and isinstance(new[key], dict)
-                ):
-                    recursively_update_map(old[key], new[key])
-                else:
-                    old[key] = new[key]
-
-        recursively_update_map(self._values, values)
+        self._values = recursively_updated_map(self._values, values)
 
     def use_values_from_yaml(self, path: Path) -> None:
         """
-        Use all values provided in a YAML/JSON file in the
-        config, override any defaults and values set by
-        previous calls into this class.
-
-        Args:
-            path (Path): Path to YAML/JSON file
+        Use values from a YAML/JSON file, overriding previous values.
         """
-
         with path.open("r") as stream:
             values = yaml.load(stream, yaml.Loader)
         self.use_values(values)
 
-    def load(self) -> C:
+    def use_values_from_env(self, prefix: str = "APP_") -> None:
         """
-        Finalize and load the config as an instance of the `schema`
-        dataclass.
+        Load values from environment variables with a given prefix.
+        """
+
+        env_values = {}
+        for key, value in os.environ.items():
+            if key.startswith(prefix):
+                # Convert key to a config path-like structure
+                config_key = key[len(prefix) :].lower()
```

Comment:
```suggestion
                config_key = key.removeprefix(prefix).lower()
```

Should this also use the same method as the CLI parsing? eg `BLUEAPI_CONFIG_FOO=bar` sets `values['config']['foo'] = "bar"` instead of relying on eval (which presumably requires something like `BLUEAPI.CONFIG='{"foo": "bar"}'`)?